### PR TITLE
Fix notebook table rendering with multiple code cells

### PR DESCRIPTION
**Pull request by lucyzhang929**
This PR addresses https://github.com/microsoft/azuredatastudio/issues/12275

Added unique query runners for each code cell to prevent updates to result tables in later code cells from affecting/changing tables in earlier code cells. 

Manually tested run all cells, add/delete cells then run again, save notebook, and change connection to make sure there are no regressions. Only change is when a new cell is run for the first time, there is a couple seconds wait time since a new query runner is being added (same wait time as running first cell in a new notebook before). 

Note: Also added missing escape import to gridOutput.component.ts that was causing some column headers to be saved incorrectly.

### CODE REVIEW

File: src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts [hunk 1]
```diff
@@ -334,8 +337,11 @@ class SqlKernel extends Disposable implements nb.IKernel {
 
 	interrupt(): Thenable<void> {
 		// TODO: figure out what to do with the QueryCancelResult
-		return this._queryRunner.cancelQuery().then((cancelResult) => {
+		let cancelPromises = [];
+		this._queryRunners.forEach(queryRunner => {
+			cancelPromises.push(queryRunner.cancelQuery().then(cancelResults => { }));
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts:342]: You should be able to simplify this to 
```
return Promise.all(this._queryRunners.map(queryRunner => queryRunner.cancelQuery())).then();
```

File: src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts [hunk 2]
```diff
@@ -265,30 +265,33 @@ class SqlKernel extends Disposable implements nb.IKernel {
...
 		let code = this.getCodeWithoutCellMagic(content);
-		if (this._queryRunner) {
+		let queryRunnerUri = 'queryRunner-' + cellUri;
+		let queryRunner: QueryRunner | undefined = this._queryRunners.get(queryRunnerUri);
+		if (queryRunner) {
 			// Cancel any existing query
-			if (this._future && !this._queryRunner.hasCompleted) {
-				this._queryRunner.cancelQuery().then(ok => undefined, error => this._errorMessageService.showDialog(Severity.Error, sqlKernelError, error));
+			if (this._future && !queryRunner.hasCompleted) {
+				queryRunner.cancelQuery().then(ok => undefined, error => this._errorMessageService.showDialog(Severity.Error, sqlKernelError, error));
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts:283]: Hm, it's interesting that we don't wait for the query to be canceled before starting the next one. @chlafreniere @corivera See any issues with that?

File: src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts [hunk 3]
```diff
@@ -159,7 +159,7 @@ export class SqlSession implements nb.ISession {
 }
 
 class SqlKernel extends Disposable implements nb.IKernel {
-	private _queryRunner: QueryRunner;
+	private _queryRunners: Map<string, QueryRunner> = new Map<string, QueryRunner>();
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts:162]: Shouldn't we be canceling/disposing the query runners when the kernel is disposed?

File: src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts [hunk 4]
```diff
@@ -265,30 +265,33 @@ class SqlKernel extends Disposable implements nb.IKernel {
 	public set connection(conn: IConnectionProfile) {
 		this._currentConnection = conn;
 		this._currentConnectionProfile = new ConnectionProfile(this._capabilitiesService, this._currentConnection);
-		this._queryRunner = undefined;
+		this._queryRunners.clear();
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts:268]: Shouldn't the previous query be canceled in this case?

**Review by Charles-Gagnon — APPROVED** [review]

File: src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts (hunk 2 above)
Comment (chlafreniere) [reply src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts:283]: Honestly don't know offhand, but I'm leery of changing this when we're close to a release.

**Review by chlafreniere — APPROVED** [review]

File: src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts [hunk 5]
```diff
@@ -372,14 +349,17 @@ class SqlKernel extends Disposable implements nb.IKernel {
 	}
 
 	public async disconnect(): Promise<void> {
-		if (this._connectionPath) {
-			if (this._connectionManagementService.isConnected(this._connectionPath)) {
-				try {
-					await this._connectionManagementService.disconnect(this._connectionPath);
-				} catch (err) {
-					this.logService.error(err);
+		if (this._queryRunners.size > 0) {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts:352]: If queryRunners length is 0 then forEach will just be a no-op so this check isn't really necessary

**Pull request merged**